### DISCUSSION TITLE
HELP Compile esp32 s3 mini with ESP-IDF V5,1.2

### DISCUSSION THREAD

**Discussion by beaulif** [revk/xESP32-Faikout#238] [Q&A]
Hi, first of all, I'm trying to flash the BIN on DIY Fakin board and Esp32 S3 Mini dev board with no luck, both are constantly rebooting. on the recommendation of RevK I tried to compile it by myself. I'm using ESP-IDF v5.1.2 but still getting errors. I'm not very familiar with esp idf. Is there anyone who can help me. Here are the errors and solutions I have

Error 1 :   Cannot find source file settings.c
 idf.py build
-- Configuring done
CMake Error at C:/Espressif/frameworks/esp-idf-v5.1.2/tools/cmake/component.cmake:479 (add_library):
  Cannot find source file:
    C:/Users/Francois/Documents/Faikin/ESP32/settings.c
  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
Call Stack (most recent call first):
  C:/Espressif/frameworks/…..

Solve: I rename C:\Users\Francois\Documents\Faikin\ESP32\settings.def by settings.c

Error 2: _signing_key
idf.py build
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Francois/Documents/Faikin/ESP32/build
[4/1087] cmd.exe /C "cd /D C:\Users\Francois\Documents\Faikin\ESP32\build && C:\Espressif\tools\cmake\3.24.0\b...sif/tools/cmake/3.24.0/bin/cmake.exe -P C:/Espressif/frameworks/esp-idf-v5.1.2/tools/cmake/scripts/fail.cmake"FAILED: CMakeFiles/gen_secure_boot_signing_key C:/Users/Francois/Documents/Faikin/ESP32/build/CMakeFiles/gen_secure_boot_signing_keycmd.ex…
Solve: idf.py menuconfig
Menu, security feature, and remove require signed app image

Error 3 : ../settings.h: No such file or directory

 idf.py build
In file included from C:/Users/Francois/Documents/Faikin/ESP32/components/ESP32-RevK/jo.c:4:
C:/Users/Francois/Documents/Faikin/ESP32/components/ESP32-RevK/include/revk.h:43:10: fatal error: ../../settings.h: No such file or directory
   43 | #include "../../settings.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

Solve no idea

**Comment by revk** [comment]
settings.c and settings.h are created using a tool in ESP32-RevK from settings.def. The Makefile should do that.

signing keys you need to create as per instructions in ESP IDF, or turn off code signing.

**Reply by beaulif** [reply]
ok but makefile does not create the files settings.c and settings.h. 
Which idf version do you use?
Compiler supported targets: xtensa-esp32s3-elf is the good one?
Please I need more information?

CDT Build Console:
Building in: C:\Users\Francois\Documents\Faikin\ESP32\build
Configuring in: C:\Users\Francois\Documents\Faikin\ESP32\build
cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=C:\Espressif\frameworks\esp-idf-v5.1.2\tools\cmake\toolchain-esp32s3.cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCCACHE_ENABLE=1 -DIDF_TARGET=esp32s3 C:\Users\Francois\Documents\Faikin\ESP32
-- Found Git: C:/Espressif/tools/idf-git/2.43.0/cmd/git.exe (found version "2.43.0.windows.1") 
-- ccache will be used for faster recompilation
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc.exe
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-g++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32s3
-- Checking Python dependencies...
Constraint file: C:\Espressif\espidf.constraints.v5.1.txt
Requirement files:
 - C:/Espressif/frameworks/esp-idf-v5.1.2/tools\requirements\requirements.core.txt
Python being checked: C:\Espressif\python_env\idf5.1_py3.11_env\Scripts\python.exe
Python requirements are satisfied.
Processing 3 dependencies:
[1/3] espressif/led_strip (2.5.3)
[2/3] espressif/mdns (1.2.4)
[3/3] idf (5.1.2)
-- Project sdkconfig file C:/Users/Francois/Documents/Faikin/ESP32/sdkconfig
Loading defaults file C:/Users/Francois/Documents/Faikin/ESP32/sdkconfig.defaults...
Compiler supported targets: xtensa-esp32s3-elf

-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of time_t
-- Check size of time_t - done
-- Found Python3: C:/Users/Francois/AppData/Local/Programs/Python/Python38-32/python.exe (found version "3.8.5") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- App "Faikin" version: 3fb90fe-dirty
-- Adding linker script C:/Users/Francois/Documents/Faikin/ESP32/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_system/ld/esp32s3/sections.ld.in
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_rom/esp32s3/ld/esp32s3.rom.ld
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_rom/esp32s3/ld/esp32s3.rom.api.ld
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_rom/esp32s3/ld/esp32s3.rom.libgcc.ld
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib.ld
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_rom/esp32s3/ld/esp32s3.rom.version.ld
-- Adding linker script C:/Espressif/frameworks/esp-idf-v5.1.2/components/soc/esp32s3/ld/esp32s3.peripherals.ld
-- Components: ESP32-RevK app_trace app_update bootloader bootloader_support bt cmock console cxx driver efuse esp-tls esp_adc esp_app_format esp_coex esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_lcd esp_local_ctrl esp_mm esp_netif esp_netif_stack esp_partition esp_phy esp_pm esp_psram esp_ringbuf esp_rom esp_system esp_timer esp_wifi espcoredump espressif__led_strip espressif__mdns esptool_py fatfs freertos hal heap http_parser idf_test ieee802154 json log lwip main mbedtls mqtt newlib nvs_flash openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs tcp_transport touch_element ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: C:/Users/Francois/Documents/Faikin/ESP32/components/ESP32-RevK C:/Espressif/frameworks/esp-idf-v5.1.2/components/app_trace C:/Espressif/frameworks/esp-idf-v5.1.2/components/app_update C:/Espressif/frameworks/esp-idf-v5.1.2/components/bootloader C:/Espressif/frameworks/esp-idf-v5.1.2/components/bootloader_support C:/Espressif/frameworks/esp-idf-v5.1.2/components/bt C:/Espressif/frameworks/esp-idf-v5.1.2/components/cmock C:/Espressif/frameworks/esp-idf-v5.1.2/components/console C:/Espressif/frameworks/esp-idf-v5.1.2/components/cxx C:/Espressif/frameworks/esp-idf-v5.1.2/components/driver C:/Espressif/frameworks/esp-idf-v5.1.2/components/efuse C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp-tls C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_adc C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_app_format C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_coex C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_common C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_eth C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_event C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_gdbstub C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_hid C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_http_client C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_http_server C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_https_ota C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_https_server C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_hw_support C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_lcd C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_local_ctrl C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_mm C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_netif C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_netif_stack C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_partition C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_phy C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_pm C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_psram C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_ringbuf C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_rom C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_system C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_timer C:/Espressif/frameworks/esp-idf-v5.1.2/components/esp_wifi C:/Espressif/frameworks/esp-idf-v5.1.2/components/espcoredump C:/Users/Francois/Documents/Faikin/ESP32/managed_components/espressif__led_strip C:/Users/Francois/Documents/Faikin/ESP32/managed_components/espressif__mdns C:/Espressif/frameworks/esp-idf-v5.1.2/components/esptool_py C:/Espressif/frameworks/esp-idf-v5.1.2/components/fatfs C:/Espressif/frameworks/esp-idf-v5.1.2/components/freertos C:/Espressif/frameworks/esp-idf-v5.1.2/components/hal C:/Espressif/frameworks/esp-idf-v5.1.2/components/heap C:/Espressif/frameworks/esp-idf-v5.1.2/components/http_parser C:/Espressif/frameworks/esp-idf-v5.1.2/components/idf_test C:/Espressif/frameworks/esp-idf-v5.1.2/components/ieee802154 C:/Espressif/frameworks/esp-idf-v5.1.2/components/json C:/Espressif/frameworks/esp-idf-v5.1.2/components/log C:/Espressif/frameworks/esp-idf-v5.1.2/components/lwip C:/Users/Francois/Documents/Faikin/ESP32/main C:/Espressif/frameworks/esp-idf-v5.1.2/components/mbedtls C:/Espressif/frameworks/esp-idf-v5.1.2/components/mqtt C:/Espressif/frameworks/esp-idf-v5.1.2/components/newlib C:/Espressif/frameworks/esp-idf-v5.1.2/components/nvs_flash C:/Espressif/frameworks/esp-idf-v5.1.2/components/openthread C:/Espressif/frameworks/esp-idf-v5.1.2/components/partition_table C:/Espressif/frameworks/esp-idf-v5.1.2/components/perfmon C:/Espressif/frameworks/esp-idf-v5.1.2/components/protobuf-c C:/Espressif/frameworks/esp-idf-v5.1.2/components/protocomm C:/Espressif/frameworks/esp-idf-v5.1.2/components/pthread C:/Espressif/frameworks/esp-idf-v5.1.2/components/sdmmc C:/Espressif/frameworks/esp-idf-v5.1.2/components/soc C:/Espressif/frameworks/esp-idf-v5.1.2/components/spi_flash C:/Espressif/frameworks/esp-idf-v5.1.2/components/spiffs C:/Espressif/frameworks/esp-idf-v5.1.2/components/tcp_transport C:/Espressif/frameworks/esp-idf-v5.1.2/components/touch_element C:/Espressif/frameworks/esp-idf-v5.1.2/components/ulp C:/Espressif/frameworks/esp-idf-v5.1.2/components/unity C:/Espressif/frameworks/esp-idf-v5.1.2/components/usb C:/Espressif/frameworks/esp-idf-v5.1.2/components/vfs C:/Espressif/frameworks/esp-idf-v5.1.2/components/wear_levelling C:/Espressif/frameworks/esp-idf-v5.1.2/components/wifi_provisioning C:/Espressif/frameworks/esp-idf-v5.1.2/components/wpa_supplicant C:/Espressif/frameworks/esp-idf-v5.1.2/components/xtensa
-- Configuring done
CMake Error at C:/Espressif/frameworks/esp-idf-v5.1.2/tools/cmake/component.cmake:479 (add_library):
  Cannot find source file:

    C:/Users/Francois/Documents/Faikin/ESP32/settings.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
Call Stack (most recent call first):
  C:/Espressif/frameworks/esp-idf-v5.1.2/tools/cmake/component.cmake:652 (idf_component_register)
  main/CMakeLists.txt:3 (register_component)


CMake Error at C:/Espressif/frameworks/esp-idf-v5.1.2/tools/cmake/component.cmake:479 (add_library):
  No SOURCES given to target: __idf_main
Call Stack (most recent call first):
  C:/Espressif/frameworks/esp-idf-v5.1.2/tools/cmake/component.cmake:652 (idf_component_register)
  main/CMakeLists.txt:3 (register_component)


CMake Generate step failed.  Build files cannot be regenerated correctly.
cmake --build . -- -v
ninja: error: loading 'build.ninja': Le fichier spécifié est introuvable.


Build complete (0 errors, 0 warnings): C:\Users\Francois\Documents\Faikin\ESP32\build
Total time taken to build the project: 27 747 ms

**Reply by revk** [reply]
It uses make, which uses a Makefile, which has a top entry dependant on settings.h which it has a target for that also makes settings.c

**Reply by beaulif** [reply]
Hi, after several hours of study and work. I made a new Linux development environment (virtualbox, ubuntu, EDF-IDE,…) because I did not succeed with Windows. Now the make and make flash commands work without error.

I still have one hardware problem that has been resolved. The 3.3v does not work I noticed that C8 does not have the right value I had a 10nf 50v instead of 10uf 35v. Do you think my 3.3v problem comes from there?
Thanks for your help.

**Reply by revk** [reply]
The cap has been wrong for a long time so won't be the issue.

**Reply by FloHomi** [reply]
Is there a way to do that on Windows?
I tried half a day and it always gets me to "csh" missing. I tried so many consoles now. Git Bash, Cygwin64 Terminal, Powershell...

**Reply by revk** [reply]
I don't have any windows machines, sorry. Some people may have managed it. The `csh` will be the scripts that update the `sdkconfig` - doing so manually or `make menuconfig` may work allowing a build using `idf.py`
